sources/packages/backend/libs/integrations/src/services/sfas/sims-to-sfas.service.ts

```diff
@@ -0,0 +1,161 @@
+  SFASBridgeLog,
+  Student,
+} from "@sims/sims-db";
+import { Brackets, Repository } from "typeorm";
+
+/**
+ * SIMS to SFAS services.
+ */
+@Injectable()
+export class SIMSToSFASService {
+  constructor(
+    @InjectRepository(Student)
+    private readonly studentRepo: Repository<Student>,
+    @InjectRepository(Application)
+    private readonly applicationRepo: Repository<Application>,
+    @InjectRepository(SFASBridgeLog)
+    private readonly sfasBridgeLogRepo: Repository<SFASBridgeLog>,
+  ) {}
+
+  /**
+   * Get the latest bridge file log date.
+   * When there is no log, null will be returned.
+   * @returns latest bridge file log date.
+   */
+  async getLatestBridgeFileLogDate(): Promise<Date | null> {
+    const [latestBridgeFileLog] = await this.sfasBridgeLogRepo.find({
+      select: { id: true, referenceDate: true },
+      order: { referenceDate: "DESC" },
+      take: 1,
+    });
+    return latestBridgeFileLog ? latestBridgeFileLog.referenceDate : null;
+  }
+
+  /**
+   * Log the details of bridge file that was sent to SFAS.
+   * @param referenceDate date when the bridge file data was extracted.
+   * @param fileName bridge file name.
+   */
+  async logBridgeFileDetails(
+    referenceDate: Date,
+    fileName: string,
+  ): Promise<void> {
+    await this.sfasBridgeLogRepo.insert({
+      referenceDate,
+      generatedFileName: fileName,
+    });
+  }
+
+  /**
+   * Get all student ids of students who have one or more updates
+   * since the date provided.
+   * The updates can be one or more of the following:
+   * - Student or User data
+   * - Sin validation data
+   * - Cas supplier data
+   * - Overawards data
+   * @param modifiedSince the date after which the student data was updated.
+   * If not provided, all the students with at least one submitted application will be returned.
+   */
+  async getAllStudentsWithUpdates(modifiedSince?: Date): Promise<number[]> {
+    const applicationStudentQuery = this.applicationRepo
+      .createQueryBuilder("application")
+      .select(["application.id", "student.id"])
+      .distinctOn(["student.id"])
+      .innerJoin("application.student", "student")
+      .innerJoin("student.user", "user")
+      .innerJoin("student.sinValidation", "sinValidation")
+      .innerJoin("student.casSupplier", "casSupplier")
+      .leftJoin("student.overawards", "overaward")
+      .where("application.applicationStatus != :overwritten", {
+        overwritten: ApplicationStatus.Overwritten,
+      })
+      .andWhere("application.currentAssessment is not null");
+    if (modifiedSince) {
+      applicationStudentQuery
+        .andWhere(
+          new Brackets((qb) => {
+            qb.where("student.updatedAt > :modifiedSince")
+              .orWhere("user.updatedAt > :modifiedSince")
+              .orWhere("sinValidation.updatedAt > :modifiedSince")
+              .orWhere("casSupplier.updatedAt > :modifiedSince")
+              .orWhere("overaward.updatedAt > :modifiedSince");
```

Comment:
Should the condition not be based on start and end dates to ensure the precise delta across all the extracted data?
We should use the last execution date (saved on DB) as a start date and set the "new execution" date (`bridgeDataExtractedDate`) at the start of the process.

Reply:
Absolutely right, I missed the point about period to ensure the integrity. Thanks. I will add it.

Reply:
Updated the logic.